lib/engine/game/g_18_norway/steps/token.rb

```diff
@@ -9,37 +9,44 @@ module Step
         class Token < Engine::Step::Token
           def available_hex(entity, hex)
             return true if super(entity, hex)
-            return true if @game.ferry_graph.reachable_hexes(entity)[hex]
+
+            return true if harbor_available?(entity, hex)
 
             false
           end
 
+          def harbor_available?(entity, hex)
+            city = @game.harbor_city_coordinates(hex.id)
+            return false if city.nil?
+
+            other_hex = @game.hex_by_id(city)
```

Comment:
I'd avoid calling a variable `city` unless you're referring to a City object. Maybe `city_hex` would be better?
Or (because the various `*_by_id` functions are implemented by hashes, you can safely pass nil or garbage values to them and expect to get nil back) you could do:
```suggestion
            other_hex = @game.hex_by_id(@game.harbor_city_coordinates(hex.id))
            return false unless other_hex

```